src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -913,15 +942,36 @@ static PyMemberDef TopicPartition_members[] = {
         { NULL }
 };
 
+static PyGetSetDef TopicPartition_getters_and_setters[] = {
```

Comment:
Why is this needed? Are we trying to restrict the user from setting this field?

Adding this only for leader_epoch should affect other attributes.